src/pa02/chutes_simulation.py

```diff
@@ -0,0 +1,276 @@
+
+        ladders: The "ladders" that should be used in the board. If none is
+                given, we use default "ladders". Ladders make player go up.
+
+        goal: The value that the player needs to reach to win the game. The
+            game stops after o ne player reaches goal.
+        """
+
+        if ladders is None:
+            ladders = Board.ladders
+        if snakes is None:
+            snakes = Board.snakes
+
+        if goal is None:
+            self.goal = Board.goal
+        else:
+            self.goal = goal
+
+        self.snakes_and_ladders = {start: end for start,
+                                   end in snakes + ladders}
+
+    def goal_reached(self, position):
+        """
+        Parameters
+        ----------
+        position: The current position of the player
+
+        return: Returns True if the position of the player have reached
+                goal or passed goal
+        -------
+        True if position is greater than end destination(goal)
+        """
+        return position >= self.goal
+
+    def position_adjustment(self, position):
+        """
+        Adjusts the position due to snakes and ladders.
+        If the player is not at a start square for snakes or ladders,
+        then it returns 0.
+
+        Parameters
+        ----------
+        position: The current position of the player
+
+        Returns
+        -------
+        The new position according to the snakes and ladders
+        """
+        new_position = self.snakes_and_ladders.get(position, position)
+
+        return new_position - position
+
+
+class Player:
+    """
+    Sets up a single player.
+    """
+    def __init__(self, board):
+        """
+        Parameters
+        ----------
+        board : The board that the player is on
+        """
+        self.board = board
+        self.position = 0
+        self.number_of_moves = 0
+
+    def move(self):
+        """
+        Moves the player to a new position
+        """
+        roll = random.randint(1, 6)
+
+        self.position += roll
+        self.position += self.board.position_adjustment(self.position)
+        self.number_of_moves += 1
+
+
+class ResilientPlayer(Player):
+    """
+    Implements a player that is more resilient.
+    This player will take x extra steps after falling down a chute.
+    """
+    def __init__(self, board, extra_steps=1):
+        """
+        Parameters
+        ----------
+        extra_steps : The number of extra steps taken after falling down a
+        snake.
+        """
+        super().__init__(board)
+        self.plus_step = extra_steps
+        self.fell_down = False
+
+    def move(self):
+        if self.fell_down:
+            extra = self.plus_step
+        else:
+            extra = 0
+
+        roll = random.randint(1, 6)
+
+        self.position += roll + extra
+        self.position += self.board.position_adjustment(self.position)
+        self.number_of_moves += 1
```

Comment:
`self.fell_down` will never become True in your code. --> The player will never take the extra steps